### PR TITLE
Add documentation support for service and resources

### PR DESCRIPTION
**Pull request by nadeeshaan**
## Approach
ezgif com-video-to-gif 3

### CODE REVIEW

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/CommandUtil.java [hunk 1]
```diff
@@ -238,12 +328,41 @@ public String getArgumentV() {
...
+    private static String getDocumentationAttribute(String docTag, String field, int offset) {
+        String offsetStr = String.join("", Collections.nCopies(offset, " "));
+        return String.format("%s%s{{%s}}", offsetStr, docTag, field);
+    }
+
+    private static String getDocumentationAttachment(List<String> attributes, int offset) {
+        String offsetStr = String.join("", Collections.nCopies(offset, " "));
+        if (attributes == null || attributes.isEmpty()) {
+            return String.format("%n%sdocumentation {%n%s\t%n%s}", offsetStr, offsetStr, offsetStr);
+        } else {
```
Comment (chamil321) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/CommandUtil.java]: Else is not required as method returns within if scope

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/CommandExecutor.java [hunk 2]
```diff
@@ -180,6 +181,20 @@ private static void executeAddDocumentation(WorkspaceServiceContext context) {
             case UtilSymbolKeys.TRANSFORMER_KEYWORD_KEY:
                 documentationContent = CommandUtil.getTransformerDocumentation(bLangPackage, line);
                 break;
+            case UtilSymbolKeys.RESOURCE_KEYWORD_KEY:
+                CommandUtil.DocAttachmentInfo attachInfo = CommandUtil.getResourceDocumentation(bLangPackage, line);
+                if (attachInfo != null) {
+                    documentationContent = attachInfo.getDocAttachment();
+                    line = attachInfo.getReplaceStartFrom();
+                }
```
Comment (chamil321) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/CommandExecutor.java]: Shall we move if scope to a granular method and reuse it below case as well?

Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/CommandExecutor.java]: This particular logic is being refactored with the new feature implementation of "Adding documentation for the whole document". This change will be included with that.

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/CommandUtil.java (hunk 1 above)
Comment (nadeeshaan) [reply language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/command/CommandUtil.java]: Will fix the same with "Adding documentation for the whole document"

**Pull request merged**